pkg/persistence/disk_persistence.go

```diff
@@ -29,11 +36,13 @@ func NewDiskHandle(path string) Handle {
 			archiveDir,
 			err,
 		)
+
+		return nil, err
```

Comment:
in this case, maybe we should just `return nil, err`? Otherwise, we would return sort of redundant message, because `createDir` already notifies that there were an error creating dir with the `dirPath`. What do you think?

Reply:
Makes sense! Please just remove `logger.Erorrf` above.